features/google/test/regex-path-matcher.test.js

```diff
@@ -0,0 +1,45 @@
+import { matchRegex } from "../src/model/importers/utils/lang-constants";
+
+describe("Test MatchRegex for File Paths", () => {
+    let german_paths_html = [
+        "Takeout/Meine Aktivitäten/Maps/MeineAktivitäten.html",
+        "Takeout/Meine Aktivitäten/Google Play Store/MeineAktivitäten.html",
+        "Takeout/Meine Aktivitäten/YouTube/MeineAktivitäten.html",
+        "Takeout/Meine Aktivitäten/Shopping/MeineAktivitäten.html",
+        "Takeout/Meine Aktivitäten/Bildersuche/MeineAktivitäten.html",
+        "Takeout/Meine Aktivitäten/Datenexport/MeineAktivitäten.html",
+        "Takeout/Meine Aktivitäten/Google Suche/MeineAktivitäten.html",
+        "Takeout/Meine Aktivitäten/Anzeigen/MeineAktivitäten.html",
+        "Takeout/Meine Aktivitäten/Google News/MeineAktivitäten.html",
```

Comment:
Are there any non-breakable spaces here? Can't see :D Would prefer using escape sequences for clarity.

Reply:
Will do that.

Reply:
Added the unicode equivalent of it, to make it explicit. Adding `\xc2\xa0` did not work. I don't know why, but it interprets each byte individually. So instead of one space I get 2 characters, some weird A and a space.

Reply:
Yeah, it's certainly non-trivial to represent a non-breakable space as surrogate pairs :D I don't think it actually works for non-astral code points. My experiments:

```
> const surrogate = (c) => [Math.floor((c - 0x10000) / 0x400) + 0xd800, (c - 0x10000) % 0x400 + 0xdc00]
undefined
> surrogate(0xa0).map((n) => n.toString(16))
(2) ['d7c0', 'd8a0']
> "\ud7c0\ud8a0"
'ퟀ\uD8A0'
> "\ud7c0\ud8a0".normalize("NFC")
'ퟀ\uD8A0'
```

So I suspect it's _not_ surrogate pairs we get, but differently normalised strings, something like NFD? That's something we could test - mix NFD and NFC normalised strings and make sure it all works out. But maybe a followup.